BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/TokenCredentialEnvelope.cs

```diff
@@ -0,0 +1,48 @@
+﻿#if NETSTANDARD2_0
+namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Authentication
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Azure.Core;
+
+    internal class TokenCredentialEnvelope : ICredentialEnvelope
+    {
+        private readonly TokenCredential tokenCredential;
+        private readonly TokenRequestContext tokenRequestContext;
+
+        public TokenCredentialEnvelope(TokenCredential tokenCredential)
+        {
+            this.tokenCredential = tokenCredential;
+            this.tokenRequestContext = new TokenRequestContext(scopes: AuthConstants.GetScopes());
+        }
+
+        public object Credential => this.tokenCredential;
+
+        public string GetToken(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var accessToken = this.tokenCredential.GetToken(requestContext: this.tokenRequestContext, cancellationToken: cancellationToken);
+            return accessToken.Token;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <remarks>
+        /// You can also use the C# default(CancellationToken) statement to create an empty cancellation token.
+        /// Source: (https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken.none).
+        /// </remarks>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<string> GetTokenAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var accessToken = await this.tokenCredential.GetTokenAsync(requestContext: this.tokenRequestContext, cancellationToken: cancellationToken);
+            return accessToken.Token;
+        }
+    }
+}
+#endif
```

Comment:
nit: put a blank line

Reply:
Thanks, I make this mistake a lot because Visual Studio doesn't call attention to it.
Are there any Analyzers that we could include in the project to fail on this?

Reply:
Perhaps steal from https://github.com/open-telemetry/opentelemetry-dotnet/blob/5f1acbf8ed360a497133b6ecae42f1268621dad6/build/sanitycheck.py#L25?